lib/dialogs/share/index.tsx

```diff
@@ -51,7 +52,7 @@ export class ShareDialog extends Component<Props> {
 
     if (collaborator !== '' && tags.indexOf(collaborator) === -1 && !isSelf) {
       this.props.editNote(noteId, { tags: [...tags, collaborator] });
-      analytics.tracks.recordEvent('editor_note_collaborator_added');
+      this.props.recordEvent('editor_note_collaborator_added');
```

Comment:
this would be an ideal place to wrap the `editNote` action with analytics middleware so that we can avoid a double-dispatch. by adding it here we make two quick state updates in succession but if we had `{ meta: { analytics: [ [ 'editor_note_collaborator_added' ] ] } }` then we could keep it to one dispatch (which mirrors the reality of all that's changing in the app) and fire the analytics in the middleware layer

```ts
if (action.meta?.analytics?.length) {
	action.meta.analytics.forEach( ( [ name, properties ] ) => analytics.tracks.recordEvent( name, properties ) )
}
```